core/runtime/v2/shim_manager.go

```diff
@@ -163,8 +165,25 @@ func (m *ShimManager) ID() string {
 
 // Start launches a new shim instance
 func (m *ShimManager) Start(ctx context.Context, id string, bundle *Bundle, opts runtime.CreateOpts) (_ ShimInstance, retErr error) {
-	// This container belongs to sandbox which supposed to be already started via sandbox API.
+	var (
+		remoteSandbox bool
+		shouldv2      bool
+	)
 	if opts.SandboxID != "" {
+		sbx, sbErr := m.sandboxStore.Get(ctx, opts.SandboxID)
```

Comment:
For context, is this sandbox store not being used in 1.X? How (or where) are sandboxes stored in those branches?

Reply:
Here, sandboxStore points to the `metadata sandbox`, which was introduced in 2.0 through the sandbox API. The creation and update of this part are done in the sandbox_run of `cri`.  see https://github.com/containerd/containerd/blob/1a43cb6a1035441f9aca8f5666a9b3ef9e70ab20/internal/cri/server/sandbox_run.go#L135C1-L137C3

`cri` create/update both cache and `metadata sandbox` data structures, which may be the reason for your confusion